yext/search-ui-react
# update VerticalConfig to Accept a Generic Type Param

This PR makes changes to VerticalConfig to accept a generic type parameter for `CardComponent `

- UniversalResults and VerticalResults have to be updated to properly infer the typing of the provided CardComponent field
- The use of `as any` is due to typing issues with result type from `State` using `useSearchState`
- The use of `any` as the generic type for StandardCard is to avoid type conflict when using it as a default value for CardComponent field in places like StandardSection
- Added some custom components for testing purposes
- Updated useCard[Blah] hooks to accept generic types for Result

J=SLAP-2233
TEST=manual

Tested VerticalResults in PeoplePage:
- use CustomCard for CardComponent field. See that the typing works as expected and no error when starting the site
- see that using StandardCard still works
Tested UniversaResults in UniversalPage:
- See that using different CustomCard of different generic type for Result in `verticalConfigMap` produce the right mapping. The typing works as expected and no error when starting the site
- See that sectionComponent pass the generic type to the CardComponent works properly using CustomSection
- see that the useCard hooks work with Result of different type in CustomCard
